sdk/src/quotes/public/decrease-liquidity-quote.ts

```diff
@@ -91,115 +91,126 @@ export function decreaseLiquidityQuoteByLiquidity(
  */
 export function decreaseLiquidityQuoteByLiquidityWithParams(
-  param: DecreaseLiquidityQuoteParam
+  params: DecreaseLiquidityQuoteParam
 ): DecreaseLiquidityQuote {
-  invariant(TickUtil.checkTickInBounds(param.tickLowerIndex), "tickLowerIndex is out of bounds.");
-  invariant(TickUtil.checkTickInBounds(param.tickUpperIndex), "tickUpperIndex is out of bounds.");
+  invariant(TickUtil.checkTickInBounds(params.tickLowerIndex), "tickLowerIndex is out of bounds.");
+  invariant(TickUtil.checkTickInBounds(params.tickUpperIndex), "tickUpperIndex is out of bounds.");
   invariant(
-    TickUtil.checkTickInBounds(param.tickCurrentIndex),
+    TickUtil.checkTickInBounds(params.tickCurrentIndex),
     "tickCurrentIndex is out of bounds."
   );
 
-  const positionStatus = PositionUtil.getStrictPositionStatus(
-    param.sqrtPrice,
-    param.tickLowerIndex,
-    param.tickUpperIndex
-  );
-
-  switch (positionStatus) {
-    case PositionStatus.BelowRange:
-      return quotePositionBelowRange(param);
-    case PositionStatus.InRange:
-      return quotePositionInRange(param);
-    case PositionStatus.AboveRange:
-      return quotePositionAboveRange(param);
-    default:
-      throw new Error(`type ${positionStatus} is an unknown PositionStatus`);
-  }
-}
-
-function quotePositionBelowRange(param: DecreaseLiquidityQuoteParam): DecreaseLiquidityQuote {
-  const { tickLowerIndex, tickUpperIndex, liquidity, slippageTolerance, tokenExtensionCtx } = param;
-
-  const sqrtPriceLowerX64 = PriceMath.tickIndexToSqrtPriceX64(tickLowerIndex);
-  const sqrtPriceUpperX64 = PriceMath.tickIndexToSqrtPriceX64(tickUpperIndex);
-
-  const tokenEstA = getTokenAFromLiquidity(liquidity, sqrtPriceLowerX64, sqrtPriceUpperX64, false);
-  const tokenMinA = adjustForSlippage(tokenEstA, slippageTolerance, false);
+  const { tokenExtensionCtx } = params;
+  const { tokenEstA, tokenEstB } = getTokenEstimatesFromLiquidity(params);
+  const [tokenMinA, tokenMinB] = [tokenEstA, tokenEstB].map((tokenEst) => adjustForSlippage(tokenEst, params.slippageTolerance, false));
 
   const tokenMinAExcluded = TokenExtensionUtil.calculateTransferFeeExcludedAmount(tokenMinA, tokenExtensionCtx.tokenMintWithProgramA, tokenExtensionCtx.currentEpoch);
   const tokenEstAExcluded = TokenExtensionUtil.calculateTransferFeeExcludedAmount(tokenEstA, tokenExtensionCtx.tokenMintWithProgramA, tokenExtensionCtx.currentEpoch);
+  const tokenMinBExcluded = TokenExtensionUtil.calculateTransferFeeExcludedAmount(tokenMinB, tokenExtensionCtx.tokenMintWithProgramB, tokenExtensionCtx.currentEpoch);
+  const tokenEstBExcluded = TokenExtensionUtil.calculateTransferFeeExcludedAmount(tokenEstB, tokenExtensionCtx.tokenMintWithProgramB, tokenExtensionCtx.currentEpoch);
 
   return {
-    liquidityAmount: liquidity,
-    tokenMinA: tokenMinAExcluded.amount,
-    tokenMinB: ZERO,
-    tokenEstA: tokenEstAExcluded.amount,
-    tokenEstB: ZERO,
+    tokenMinA,
```

Comment:
hmm, if TransferFee is enabled, `tokenMin` and `tokenEst` will be `original amount - transfer fee`.
So original code sounds good to me.
WDYT ?